### DISCUSSION TITLE
Hub utils

### DISCUSSION THREAD

**Discussion by dalechyn** [wevm/frog-archived#27] [Ideas]
`frame.js` offers [`getAddressForFid`](https://framesjs.org/reference/js/getAddressForFid) and [`getUserDataForFid`](https://framesjs.org/reference/js/getUserDataForFid).

Would it make sense to have similar in `frog`? Or is this rather not `frog`'s responsibility to provide hub helpers?

**Comment by tmm** [comment]
Maybe at some point, but we are more focused on Frame app structure and "rendering" at the moment.

Frames.js, [Neynar](https://neynar.com), or another Farcaster data library is very complementary to Frog right now. This could be a great candidate for [Frog Middleware](https://warpcast.com/awkweb/0xd362d419) at some point too.
